aws-rds-dbinstance/src/main/java/software/amazon/rds/dbinstance/CreateHandler.java

```diff
@@ -210,10 +216,29 @@ private String fetchEngine(final ProxyClient<RdsClient> client, final ResourceMo
         }
 
         if (ResourceModelHelper.isDBInstanceReadReplica(model)) {
-            return fetchDBInstance(client, model.getSourceDBInstanceIdentifier()).engine();
+            final String sourceDBInstanceArn = model.getSourceDBInstanceIdentifier();
+            final String sourceDBInstance = ResourceModelHelper.isValidArn(sourceDBInstanceArn) ?
+                    ResourceModelHelper.getResourceNameFromArn(sourceDBInstanceArn) : sourceDBInstanceArn;
+            if (ResourceModelHelper.isCrossRegionDBInstanceReadReplica(model, currentRegion)) {
+                final String sourceRegion = ResourceModelHelper.getRegionFromArn(sourceDBInstanceArn);
```

Comment:
Just a few lines above we are checking if the ARN is valid. What happens if a malformed ARN is sent to the method?

Reply:
`isCrossRegionDBInstanceReadReplica` function is to validate the correctness of an ARN and retrieve its region. This region then compared with the current region. If the comparison yields false, the logic will not continue. Therefore, if the ARN is malformed, this logic will not be executed.